### PR TITLE
refactor: Modify Ui to be static

### LINKED ISSUE #72: refactor: Modify Ui to be static

### PR DESCRIPTION
**Pull request by nickczh**
# Checklist for feature implementation and modification
Closes #72 
- [x] Added relevant milestone (under milestone tab on the right)
- Implementation
  - [ ] Sources (if any) credited under both relevant section(s) and Developer Guide (Acknowledgement)
  - [ ] JUnit Test(s)
- Documentation
  - [ ] JavaDoc for classes and methods
  - Developer Guide
    - [ ] Design (Can add as a todo statement for later on)
    - [ ] Implementation (Can add as a todo statement for later on)
    - [ ] Relevant User Story present
    - [ ] Non-functional requirement(s) that the feature addresses (if any)
    - [ ] Glossary of the (technical) term(s) used (if any)
    - [ ] Specialised instructions for manual testing (if any)
  - User Guide
    - [ ] Description and usage of feature (if any)
  - Contributions (under `/docs/team/*.md`)
    - [ ] Code/Feature/Enhancement contribution
    - [ ] Developer Guide contribution
    - [ ] User Guide contribution
    - [ ] Misc and team-based admin tasks (if any)

# Checklist for non-implementation related PR
Closes #72 
- [x] Added relevant milestone (under milestone tab on the right)
- Documentation (Contributions)
  - [ ] Developer Guide contributions (if any)
  - [ ] User Guide contributions (if any)
  - [ ] Review contributions (if any)
  - [ ] Miscellaneous and team-based admin tasks (if any)

# Additional Information

Nil

### CODE REVIEW

File: src/main/java/meditracker/MediTracker.java [hunk 1]
```diff
@@ -17,15 +17,13 @@
  */
 public class MediTracker {
 
-    private Ui ui;
     private MedicationManager medicationManager;
     private DailyMedicationManager dailyMedicationManager;
 
     /**
      * Constructs a new MediTracker object and initializes the user interface.
```
Comment (e0958902) [inline src/main/java/meditracker/MediTracker.java:24]: Perhaps, you could update this JavaDoc comment too as Ui is now static.
The rest of the JavaDoc comments were updated accordingly.

File: src/main/java/meditracker/command/SearchCommand.java [hunk 2]
```diff
@@ -10,7 +10,7 @@
 /**
  * Represents a command to search for medications from the local medication library based on a keyword.
  */
-public class SearchCommand extends Command{
+public class SearchCommand extends Command {
```
Comment (e0958902) [inline src/main/java/meditracker/command/SearchCommand.java:13]: Good catch on adding a space before the start of the curly brackets.

**Review by e0958902 — CHANGES_REQUESTED** [review]
I could only spot the JavaDoc comment. The rest looks good to me!

File: data/dailymed/today.txt
Comment (wenenhoe) [inline data/dailymed/today.txt:1]: Please remove this file and add to `.gitignore` as well so it is not checked into the repository

File: text-ui-test/data/dailymed/today.txt
Comment (wenenhoe) [inline text-ui-test/data/dailymed/today.txt:1]: Please remove this file and add to `.gitignore` as well so it is not checked into the repository

**Review by wenenhoe — CHANGES_REQUESTED** [review]
Overall, good job on refactoring the `Ui` class to be static. I have added some comments on removing some of the data files and preventing them from being checked into the repository, do take a look at them.

File: data/dailymed/today.txt
Comment (nickczh) [reply data/dailymed/today.txt:1]: Resolved in 6e84b1.

File: src/main/java/meditracker/command/SearchCommand.java (hunk 2 above)
Comment (nickczh) [reply src/main/java/meditracker/command/SearchCommand.java:13]: Thank you! ☺️

File: src/main/java/meditracker/MediTracker.java (hunk 1 above)
Comment (nickczh) [reply src/main/java/meditracker/MediTracker.java]: Resolved in https://github.com/AY2324S2-CS2113T-T09-1/tp/pull/73/commits/44c762e3450fa42e124449b4bd626bc60faceff5

**Review by wenenhoe — APPROVED** [review]
LGTM

**Review by e0958902 — APPROVED** [review]
LGTM!

**Pull request merged**